File: src/main/java/toast/pine/Classes/Items/ItemListener.java
package toast.pine.Classes.Items;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.inventory.ItemStack;
import org.bukkit.persistence.PersistentDataContainer;
import toast.pine.Events.PlayerLeftClickEvent;
import toast.pine.Events.PlayerRightCLickEvent;
import toast.pine.Keys;
import toast.pine.ToastRPG;

public class ItemListener implements Listener {

    private final ItemManager manager;

    public ItemListener(ItemManager manager) {
        this.manager = manager;
    }

    @EventHandler
    public void onRightClick(PlayerRightCLickEvent event) {
        ItemStack item = event.getMainHand();
        PersistentDataContainer container = item.getItemMeta().getPersistentDataContainer();
        if (container.has(Keys.ITEM)) {
            Item itemObject = container.get(Keys.ITEM, ToastRPG.getAdapterManager().getItemAdapter());
            this.manager.handleRightClick(itemObject, event);
        }

    }

    @EventHandler
    public void onLeftClick(PlayerLeftClickEvent event) {
        ItemStack item = event.getMainHand();
        PersistentDataContainer container = item.getItemMeta().getPersistentDataContainer();
        if (container.has(Keys.ITEM)) {
            Item itemObject = container.get(Keys.ITEM, ToastRPG.getAdapterManager().getItemAdapter());
            this.manager.handleLeftClick(itemObject, event);
        }
    }


}

File: src/main/java/pine/toast/Classes/Items/ItemListener.kt
package toast.pine.Classes.Items

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.inventory.ItemStack
import org.bukkit.persistence.PersistentDataContainer
import toast.pine.Events.PlayerLeftClickEvent
import toast.pine.Events.PlayerRightCLickEvent
import pine.toast.Keys
import pine.toast.ToastRPG

class ItemListener(private val manager: ItemManager) : Listener {

    @EventHandler
    fun onRightClick(event: PlayerRightCLickEvent) {
        val item: ItemStack = event.getMainHand()
        val container: PersistentDataContainer = item.itemMeta.persistentDataContainer
        if (container.has(Keys.ITEM)) {
            val itemObject: Item? = ToastRPG.getAdapterManager()?.let { container.get(Keys.ITEM, it.itemAdapter) }
            if (itemObject != null) {
                manager.handleRightClick(item, event)
            }
        }
    }

    @EventHandler
    fun onLeftClick(event: PlayerLeftClickEvent) {
        val item: ItemStack = event.getMainHand()
        val container: PersistentDataContainer = item.itemMeta.persistentDataContainer
        if (container.has(Keys.ITEM)) {
            val itemObject: Item? = ToastRPG.getAdapterManager()?.let { container.get(Keys.ITEM, it.itemAdapter) }
            if (itemObject != null) {
                manager.handleLeftClick(item, event)
            }
        }
    }
}
